### PR TITLE
Display mentors info on mentors tab

### PR DESCRIPTION
**Pull request by tsleal29**

### CODE REVIEW

**Comment by tsleal29** [comment]
> Next step would be to:
> 
> * add validations to the model
>   
>   * 1 < xp < 4
>   * location is either Nantes, Berlin or Paris
> * allow users to change their information

Validations: done!
Allow user to change their info: WIP

File: db/migrate/20200811190846_add_mentor_info_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class AddMentorInfoToUsers < ActiveRecord::Migration[6.0]
+  def change
+    change_table :users do |t|
+      t.integer "location"
```
Comment (fwuensche) [inline db/migrate/20200811190846_add_mentor_info_to_users.rb:4]: you're following the rails convention, that's perfect. but at doctolib we use a string to store enums. it's helpful when querying the db, and avoids confusion between numbers when we add new values. also, we're using uniq to train new joiners at doctolib, so it'd be nice to do it like that. i'll leave another comment to show what it looks like in the model.

File: app/models/user.rb [hunk 2]
```diff
@@ -8,6 +8,14 @@ class User < ApplicationRecord
   belongs_to :mentor, class_name: 'User', optional: true
   has_many :user_lessons
 
+  enum location: %w[ Paris Nantes Berlin ]
```
Comment (fwuensche) [inline app/models/user.rb:11]: image

also, make sure you activate rubocop locally. I believe it'd fail in this line.

File: db/migrate/20200811190846_add_mentor_info_to_users.rb (hunk 1 above)
Comment (tsleal29) [reply db/migrate/20200811190846_add_mentor_info_to_users.rb:4]: I have to admit I didn't like the integer solution very much, but followed the convention in the doc. I do prefer the string solution 👍

File: app/models/user.rb (hunk 2 above)
Comment (tsleal29) [reply app/models/user.rb]: > image
> 
> also, make sure you activate rubocop locally. I believe it'd fail in this line.

Done 👌

File: app/models/user.rb [hunk 3]
```diff
@@ -8,8 +8,20 @@ class User < ApplicationRecord
...
+  }
+
+  validates :rails_xp, :react_xp, numericality: {
+    only_integer: true, allow_nil: true,
+    greater_than_or_equal_to: 1,
+    less_than_or_equal_to: 4
+  }
+
   def mentor?
-    !!User.find_by(mentor_id: self.id)
```
Comment (fwuensche) [inline app/models/user.rb:12]: Beautiful scouting

File: app/models/user.rb [hunk 4]
```diff
@@ -27,7 +39,7 @@ def doctoliber?
   end
 
   def self.from_omniauth(auth)
-    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+    where(provider: auth.provider, uid: auth.uid).first_or_create! do |user|
```
Comment (fwuensche) [inline app/models/user.rb:42]: Thanks! Do you understand why?

File: app/models/user.rb [hunk 5]
```diff
@@ -8,8 +8,20 @@ class User < ApplicationRecord
   belongs_to :mentor, class_name: 'User', optional: true
   has_many :user_lessons
 
+  enum location: {
+    paris: 'paris',
+    nantes: 'nantes',
+    berlin: 'berlin',
+  }
+
+  validates :rails_xp, :react_xp, numericality: {
```
Comment (fwuensche) [inline app/models/user.rb:17]: Pretty neat

**Pull request merged**